pkg/metadata/inventories/host_metadata.go

```diff
@@ -107,17 +107,18 @@ func getHostMetadata() *HostMetadata {
 		metadata.CPUCacheSize = cpuInfo.CacheSizeBytes
 	}
 
-	platformInfo, warnings, err := platformGet()
+	platformInfo := platformGet()
+	_, warnings, err = platformInfo.AsJSON()
 	if err != nil {
 		logErrorf("failed to retrieve host platform metadata from gohai: %s", err) //nolint:errcheck
 	} else {
 		logWarnings(warnings)
 
-		metadata.KernelName = platformInfo.KernelName
-		metadata.KernelRelease = platformInfo.KernelRelease
-		metadata.KernelVersion = platformInfo.KernelVersion
-		metadata.OS = platformInfo.OS
-		metadata.CPUArchitecture = platformInfo.HardwarePlatform
+		metadata.KernelName, _ = platformInfo.KernelName.Value()
```

Comment:
What do you think about introducing a `ValueOrDefault` method? It is uncommon to rely on the default value when there is an error.

Reply:
I guess it would make this kind of use clearer and more explicit so it's a good idea, thanks for the suggestion !

I think I should do that in a separate PR, since it would be used in `memory`, `platform` and `cpu` (and any future new api for a package), what do you think ?